improver/blending/spatial_weights.py

```diff
@@ -307,21 +303,28 @@ def process(self, cube_to_collapse, one_dimensional_weights_cube, blend_coord):
                 name given blend_coord. This cube contains 1D weights
                 that will be applied along the blend_coord but need
                 adjusting spatially based on missing data.
-            blend_coord (str):
-                A string containing the name of the coordinate that the
-                cube_to_collapse will be collapsed along. Also matches the
-                coordinate in one_dimensional_weights_cube.
 
         Returns:
             iris.cube.Cube:
-                A cube containing non-normalised spatial weights based on the
-                cube_to_collapsemask and the one_dimensional weights supplied.
-                Contains the dimensions, blend_coord, y, x.
+                A cube containing normalised spatial weights based on the
+                cube_to_collapse mask and the one_dimensional weights supplied.
+                Has dimensions: self.blend_coord, y, x.
         """
-        template_cube = self.create_template_slice(cube_to_collapse, blend_coord)
-        weights_from_mask = self.create_initial_weights_from_mask(template_cube)
-        weights_from_mask = self.smooth_initial_weights(weights_from_mask)
-        final_weights = self.multiply_weights(
-            weights_from_mask, one_dimensional_weights_cube, blend_coord
-        )
-        return final_weights
+        template_cube = self._create_template_slice(cube_to_collapse)
+        (self.blend_axis,) = template_cube.coord_dims(self.blend_coord)
+
+        weights = self._broadcast_weights(template_cube, one_dimensional_weights_cube)
+
+        if np.ma.is_masked(template_cube.data):
+            # Set masked weights to zero
+            weights.data = np.where(template_cube.data.mask, 0, weights.data)
+        else:
+            message = "Expected masked input to SpatiallyVaryingWeightsFromMask"
+            warnings.warn(message)
+            return weights
+
+        weights = self._normalise_initial_weights(weights)
+        weights, rescaled = self._rescale_masked_weights(weights)
+        self._rescale_unmasked_weights(weights, rescaled)
```

Comment:
When we use masked arrays in a normal (i.e. non-fuzzy-weight) environment, we can sometimes get away with not normalising here, as the weights get normalised within the np.ma operations (e.g. mean). I think, however, because of the operations here, we do need to normalise initially, as you have done.

I wondered if you should put a comment in just saying why the normalisation has to happen here rather than relying on np.ma operations.

Reply:
I could elaborate in the "process" docstring - that's probably the best place to put this.